src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -38,6 +38,9 @@ public RegionInfo(string region, RegionAutodetectionSource regionSource, string
         private readonly IHttpManager _httpManager;
         private readonly int _imdsCallTimeoutMs;
 
+        // lock for discovery. So it is done only once
+        private readonly SemaphoreSlim _lockDiscover = new SemaphoreSlim(1);
```

Comment:
We need to unit test this change, i.e. 

1. start request1 and make it pause inside the crtical section (for example by adding a pause in the http manager)
2. start request2 and ensure it is blocked waiting
3. complete request1 and assert request2 does not make HTTP request.

Reply:
Unit test added